templates/stats.ejs

```diff
@@ -1 +1,324 @@
+                // TODO: Have a discussion on whether this may be the better approach
+                // function initializeAuxiliaryProperty(stat_property) {
+                //     var stat_value = locals[stat_property] ? locals[stat_property] : default_value;
+                //     return stat_value;
+                // }
+                // var stats_aux = {}
+                // for (stat of stats_variables) {
+                //     stats_aux[stat + "_aux"] = initializeAuxiliaryProperty(stat);
+                // }
+                // delete stats_aux["v_max_stat_aux"];
+                console.log(locals);
+
+                function initializeAuxiliaryProperty(stat_property, default_value) {
+                    var stat_value = locals[stat_property] && !Number.isNaN(locals[stat_property]) ? 
+                        parseInt(locals[stat_property]) : 0;
+                    return stat_value;
+                }
+                var stats_aux = {
+                    "v_available_points_aux": initializeAuxiliaryProperty("v_available_points", 0),
+                    "v_strength_aux": initializeAuxiliaryProperty("v_strength", 0),
+                    "v_toughness_aux": initializeAuxiliaryProperty("v_toughness", 0),
+                    "v_agility_aux": initializeAuxiliaryProperty("v_agility", 0),
+                    "v_reflexes_aux": initializeAuxiliaryProperty("v_reflexes", 0),
+                    "v_hearing_aux": initializeAuxiliaryProperty("v_hearing", 0),
+                    "v_perception_aux": initializeAuxiliaryProperty("v_perception", 0),
+                    "v_ancient_languages_aux": initializeAuxiliaryProperty("v_ancient_languages", 0),
+                    "v_combat_technique_aux": initializeAuxiliaryProperty("v_combat_technique", 0),
+                    "v_premonition_aux": initializeAuxiliaryProperty("v_premonition", 0),
+                    "v_bluff_aux": initializeAuxiliaryProperty("v_bluff", 0),
+                    "v_magical_sense_aux": initializeAuxiliaryProperty("v_magical_sense", 0),
+                    "v_aura_hardening_aux": initializeAuxiliaryProperty("v_aura_hardening", 0),
+                    "v_magical_power_aux": initializeAuxiliaryProperty("v_magical_power", 0),
+                    "v_magical_knowledge_aux": initializeAuxiliaryProperty("v_magical_knowledge", 0),
+                }
+                return stats_aux;
+            }
+
+            // This function is called on page load to initialize the stats
+            // As well as to update when Cancel Changes is clicked
+            // Unlike getAuxiliaryStats, this function is implemented in a flexible manner, instead of hard-coding
+            //  For example: notice the calculation of stat, stat_value, stat_field_id, stat_value_span_id
+            // TODO: Have a discussion on whether this may be too flexible and if it is better to hard-code the stats
```

Comment:
So after a lot of back and forth, I ended up making some functions very flexible like `initializeStats`, but some functions with full hard-coding, like `getAuxiliaryStats` (although i have added a flexible version of that same function in comments). 

Feel free to let me know which approach y'all are comfortable with. 
I am, for a change, more inclined towards hard-coding as this page is going to be relatively consistent throughout the 5 books.

Reply:
If you are fine with how it is, then we can merge it. 
It ideally, wouldn't take long to make the whole file fully flexible or mostly hard-coded. We can do that in a separate PR.